pkg/registry/apis/datasource/sub_query.go

```diff
@@ -0,0 +1,103 @@
+package datasource
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/registry/rest"
+)
+
+type subQueryREST struct {
+	builder *DataSourceAPIBuilder
+}
+
+var _ = rest.Connecter(&subQueryREST{})
+
+func (r *subQueryREST) New() runtime.Object {
+	return &metav1.Status{}
+}
+
+func (r *subQueryREST) Destroy() {
+}
+
+func (r *subQueryREST) ConnectMethods() []string {
+	return []string{"POST", "GET"}
+}
+
+func (r *subQueryREST) NewConnectOptions() (runtime.Object, bool, string) {
+	return nil, false, ""
+}
+
+func (r *subQueryREST) readQueries(req *http.Request) ([]backend.DataQuery, error) {
+	// Simple URL to JSON mapping
+	if req.Method == http.MethodGet {
+		body := make(map[string]any, 0)
```

Comment:
This is a *very* basic connector to support query over GET, we can expand on this,


* [/query?scenarioId=random_walk_table](http://localhost:3000/apis/testdata.datasource.grafana.app/v0alpha1/namespaces/default/connections/PD8C576611E62080A/query?scenarioId=random_walk_table)
* [/query?scenarioId=random_walk](http://localhost:3000/apis/testdata.datasource.grafana.app/v0alpha1/namespaces/default/connections/PD8C576611E62080A/query?scenarioId=random_walk)